Misc Tweaks

Chris--B opened the pull request:
- Added clap
- Added Debug impls for types
- Added tracing to visualize where time is spent

Chris--B committed:
Switch from Gumdrop to Clap for CLI

Clap is more powerful and the build delta is marginal. I saw approx.
15 seconds for both versions.

Chris--B committed:
Fix scaling issue in ImGui on MacOS

The drawing viewport for the ImGui renderer wasn't taking into
account the hidpi scaling factor. ImGui already tells us about it
with a `framebuffer_scale` member on the draw data. (It was already
used for the clipping logc).

impakt73 commented on src/bin/sim.rs:
```diff
@@ -1,21 +1,23 @@
+
+    /// Path to write out the framebuffer as a png
+    #[clap(short='o')]
+    framebuffer_path: Option<String>,
```
why not name this image_path to be consistent with the code below?

impakt73 commented on src/protobridge.rs:
```diff
@@ -18,13 +18,14 @@ struct DataStatus {
 
 extern "C" {
-    fn CreateProtoBridge(p_handle: *mut ProtoBridgeHandle) -> u32;
+    fn CreateProtoBridge(p_handle: &mut ProtoBridgeHandle) -> u32;
```
if this is called p_handle why is it a ref now?

Chris--B committed:
Minor tweaks to Cargo.toml and gitignore

This sorts the dependencies to make them easier to read and adds a
few patterns to `.gitignore`. This unbreaks my `cargo-watch`
workflow. It was re-triggering a lot because it creates new files.

Chris--B committed:
cargo fmt + review feedback
